### ISSUE TITLE
Testset

### ISSUE THREAD

**Issue by VisionLover** [haoningwu3639/StoryGen#32]
Thank you for recently providing the test set files. I have a few questions regarding them:

1. Do these files require the Data Processing Pipeline?

2. In the **_dataset.py_** file, you defined the **_StorySalonDataset_** class for the training data, which includes **_prev_images_** and **_prev_prompts_**. Is there an equivalent file for the test set, or do we need to create something similar ourselves?

**Comment by haoningwu3639** [comment]
1. These data have been processed by the data processing pipeline and can be directly used for model performance testing.
2. Please refer to the experimental section and appendix of our paper. When conducting quantitative evaluation, we use one frame of prev_image and prev_prompt as conditions to generate the current frame. You should change the 3 frames in dataset.py to the corresponding 1 frame as conditions.

**Comment by VisionLover** [comment]
Thank you for your reply. I have two more questions:

In the experiments mentioned in the paper, there are no results for the **_PororoSV_** and **_FlintstonesSV_** datasets. Have you tested your model on these two datasets?
While examining your training dataset, I noticed that many _prev_image_ and _prev_prompts_ entries do not seem to be related to the current prompt (as shown in the example below). What is the reason for this discrepancy, and will this lack of coordination affect the results?
Thank you for your attention and response.
<img width="802" alt="Annotation 2024-07-22 153423" src="https://github.com/user-attachments/assets/ff7310f5-9412-47f0-9de5-8cf8713fcc14">

@haoningwu3639

**Comment by haoningwu3639** [comment]
1. We did not test on the PororoSV and FlintstonesSV datasets. This is because our goal is ‘open-ended visual storytelling’, and the above datasets have only about 10 characters, which cannot meet the training requirements of our task. It would be somehow unfair to directly zero-shot compare our StoryGen with the model trained on these data, but during the writing of the paper, we did not consider fine-tuning on these data.
2. 2. For our test set, we randomly sampled from the entire StorySalon dataset to ensure fairness. Therefore, the above situation sometimes occurs when only one previous frame is used as condition. Although it cannot provide the character's appearance, the previous frame can still provide a reference for style. Honestly, it isn't easy to obtain a coherent story with specific characters with public copyright.
